# Lab #2: Democrats 2018 Primary Results and Data Analysis

Lillian Ellis

Created September 11, 2018

**Abstract:**

In this lab, I worked with a data set surrounding the democrats involved in the primary elections this year around our country. Throughout the lab, I worked to answer a few questions. How many candidates were running for governor? What about the U.S. House of Representatives? How many identified as LGBTQ, non-white, or were veterans? How many were Obama Alum? How many of these democratic candidates advanced to the general election? How many lost their races?

**Data Set Preparation:**

My data is all from FiveThirtyEight: a well known, reputable website founded by Nate Silver that focuses on data analysis, specifically poll analysis. The data set contains information from all open democratic primary elections through August 7th of 2018. The data set contains information on the 811 democratic candidates who ran for various positions up until the 7th. The specific data set I used contained 812 lines of data, and was extremely well organized. One downside of the data-set was that some information surrounding party support, Emily endorsed, gun sense, Biden endorsed, and more were missing or partially filled out. The table was not fully complete. However, the data was well organized, accurate, and the GitHub description contained information on where each number, fact, or name was pulled from.

Link to Data Set (From FiveThirtyEight's GitHub): https://github.com/fivethirtyeight/data 

Link to Five Thirty Eight Article using Data: https://fivethirtyeight.com/features/the-establishment-is-beating-the-progressive-wing-in-democratic-primaries-so-far/

Here, I opened the file, and seperated each row by commas using the ".split" method. By splitting each row in this way, I created a list of lists to find under the list "candidate_data" which were split by rows. Unlike lab one, where I created a for loop to append them all to lists by row, here I am making a literal list of lists by row. This will make it extremley easy to pull information from a specific candidate, as I do not have to search for an index as I did in Lab 1, but can pull from my "list of lists." 

I then closed the file, and deleted the first 3 rows (using :2, as they were all titles)

I then defined the column's indexes into variables for organizational purposes (so: candidate name, state name, district, etc.)

In [4]:
#opening file
data_file = open("dem_candidates.csv", "r")

#creating list to store the data
candidate_data = [];

#appending data from file into the list

for row in data_file:
    candidate_data.append(row.split(",")); 
    
#closing file
data_file.close();

#remove the first element (column header)
del candidate_data[:2];


#defining all columns indexes under variables 
candidate_name = 0
state_name = 1
district = 2
office_type = 3
race_type = 4
election_data = 5
primary_status = 6
primary_runoff = 7
general_status = 8
partisan_lean = 9
primary_pct = 10
won_primary = 11
candidate_race = 12
veteran = 13
LGBTQ = 14
elected_official = 15
self_funder = 16
STEM = 17
Obama_alum = 18


**Data Modeling:**

To gather all of this information, I mainly used a for loop to gather data in my "list of lists: candidate_data." I then put an if statement in this for loop which looked for the positive which I was trying to gather. For this, I used i as a way to check every element, and the defined index column which I defined above. I then counted this with the count + 1 to continue the loop to cycle through all of the data, and then implemented a print statement to print the findings.  

In [5]:

print("Candidate Information:")

print(len(candidate_data),"candidates ran this past primary race")
#count for governor 
gov_count = 0

for i in range (len(candidate_data)):
    if candidate_data[i][office_type] == "Governor":
        gov_count = gov_count + 1

print(gov_count, "candidates ran for governor")

#for count in US count of representatives 
us_rep = 0
for i in range (len(candidate_data)):
    if candidate_data[i][office_type] == "Representative":
        us_rep = us_rep + 1
print(us_rep, "candidates ran for U.S. House of Representatives")

#counting how many candidates lost 
loss_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][primary_status] == "Lost":
        loss_count = loss_count + 1
print(loss_count, "democratic candidates lost")

#counting how many candidates won
win_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][primary_status] == "Advanced":
        win_count = win_count + 1
print(win_count, "democratic candidates advanced to the general election")

#counting how many LGBTQ candidates ran overall
LGBTQ_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][LGBTQ] == "Yes":
        LGBTQ_count = LGBTQ_count + 1
print(LGBTQ_count, "LGBTQ democratic candidates ran for a position this past primary")

#counting how many non-white candidates ran overall
non_white_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][candidate_race] == "Nonwhite":
        non_white_count = non_white_count + 1
print(non_white_count, "non-white democratic candidates ran this past primary")

#print how many candidates are Obama alum 
obama_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][Obama_alum] == "Yes":
        obama_count = obama_count + 1
print(obama_count, "candidates who ran are Obama Alum")

#print how many candidates are veterans
veteran_count = 0
for i in range (len(candidate_data)):
    if candidate_data[i][veteran] == "Yes":
        veteran_count = veteran_count + 1
        
print(veteran_count, "democratic candidates are veterans")
    
        

Candidate Information:
811 candidates ran this past primary race
100 candidates ran for governor
687 candidates ran for U.S. House of Representatives
546 democratic candidates lost
265 democratic candidates advanced to the general election
37 LGBTQ democratic candidates ran for a position this past primary
203 non-white democratic candidates ran this past primary
34 candidates who ran are Obama Alum
123 democratic candidates are veterans


Next, I wanted to count the individual, unique amount of states each candidate represented. To do this, I used the set function to count unique names, so the state of Alabama for example wasn't counted more than once.  

In [6]:
#candidate state number
set_length = (len(set(candidate_data[state_name])))
print("There were", set_length, "different states where candidates ran")

There were 15 different states where candidates ran


**Data Analysis and Conclusion**

Overall, I learned alot from the data set and analysis at hand. Out of the 811 democratic candidates that this data set covered, around 12% of them ran for governor, while around 84% ran for U.S. House of Representatives. The majority of candidates lost thier race, but the winning democrat moves onto the general election- as the 265 advancing are the democratic candidate in the general. 4.6% of candidates are LGBTQ, and 25% of them do not identify as white. About 15% of candidates who ran are veterans. A large amount of democratic candidates ran for office this year, as more candidates lost than one (as they were racing against eachother). This should mean a good turnout of voters and support for the general, as the democratic winner was narrowed against a tougher competition with more candidates. This large show of democratic candidates suggests a strong support for democrats in November, but could also lead to a split vote as more candidates ran and a greater variety were supported. In addition, this year a fair amount of LGBTQ, non-white, and veterans ran for office, making for a more diverse and different group of candidates. 

**Acknowledgements**

For this project, I referred to the More Python Practice file that Ms. Sconyers shared with us to recall certain functions for calling, reading, and organizing data. I worked with Ms. Sconyers alot on how to organize my data. Additionally, I used John Gruber's Canonical Markdown Cheatsheet. All data was pulled from Five Thirty Eight.  